### ISSUE TITLE
Can't connect to the AP

### ISSUE THREAD

**Issue by jonas-w** [toddaustin07/rpi-st-device#6]
Even in the testing phase where the Hotspot is called "MyPiTestAccessPoint" or something like that i can't connect. My phone only says "Network error" and the log from hostapd only shows that a handshake is started and then the connection is lost!
Any ideas?

**Comment by toddaustin07** [comment]
You won't be able to connect to MyPiTestAccessPoint with your phone.  You can only use that to verify that the AP is functioning.

If you want to set up your Pi to be a general purpose AP that you can connect to from your phone, and access the internet, etc., there is a way to do that but it takes some additional configuration of the Pi.

**Comment by jonas-w** [comment]
Oh okay i thought this would be possible.

Then this is probably another issue.

The device pops up in SmartThings but after waiting a long time it fails and tells me to connect manually which would require to connect to the AP manually. 

What logs would you need to debug this issue?
I've used your script and didn't change anything in the config except the WIFI Country

### Should also not that I don't have a desktop environment installed

**Comment by jonas-w** [comment]
#### Hostapd output:
```
	Starting example application...


Goodbye

pi@hostname:~/rpi-st-device $ I (1656022889:846) [IoT]: _dump_device_info(706) > firmware_version: switch_example_001
I (1656022889:847) [IoT]: iot_bsp_wifi_init(149) > [rpi] iot_bsp_wifi_init
I (1656022889:847) [IoT]: iot_bsp_wifi_init(150) > [rpi] wifi module version 20210405
option parsing failed: Cannot open display:
I (1656022889:989) [IoT]: iot_bsp_wifi_init(164) > [rpi] Ethernet connection available: eth0
I (1656022889:989) [IoT]: iot_bsp_wifi_init(182) > [rpi] Wifi AP device wlan0 found
I (1656022889:989) [IoT]: iot_bsp_wifi_init(230) > [rpi] Wifi Initialization Done
I (1656022889:989) [IoT]: st_conn_init(1372) > previous error code[]
I (1656022889:989) [IoT]: st_conn_init(1380) > stdk_version : 1.7.5
I (1656022889:989) [IoT]: st_conn_start(1850) > st_conn_start start (no-pin)
I (1656022889:990) [IoT]: _do_iot_main_command(498) > curr_main_cmd:10, curr_main_state:-1/-1
I (1656022889:990) [IoT]: _do_update_timeout(199) > Current timeout : 100000 for -1/0
I (1656022889:990) [IoT]: _do_status_report(304) > Call usr status_cb with 1/0
iot_status: 1, lv: 0
I (1656022889:990) [IoT]: _do_iot_main_command(498) > curr_main_cmd:2, curr_main_state:-1/0
E (1656022889:990) [IoT]: _iot_security_be_bsp_fs_load_from_nv(129) > iot_bsp_fs_open(./WifiProvStatus) = -1208
I (1656022889:990) [IoT]: _do_cmd_tout_check(1041) > New state updated for 0
I (1656022889:990) [IoT]: _do_iot_main_command(498) > curr_main_cmd:10, curr_main_state:0/0
I (1656022889:990) [IoT]: iot_bsp_wifi_set_mode(871) > [rpi] Requested mode SCAN
I (1656022889:990) [IoT]: _perform_scan(1613) > [rpi] Running Wifi AP scan using wlan0
I (1656022890:692) [IoT]: iot_bsp_wifi_set_mode(889) > [rpi] WiFi scan completed. 20 APs found
I (1656022890:692) [IoT]: iot_bsp_wifi_get_scan_result(1587) > [rpi] Get scan result requested; 20 available
I (1656022890:692) [IoT]: iot_easysetup_create_ssid(89) > >> Switch Test[s9an] <<
I (1656022890:692) [IoT]: _do_update_timeout(199) > Current timeout : 300000 for 0/1
I (1656022890:693) [IoT]: _do_iot_main_command(498) > curr_main_cmd:1, curr_main_state:0/1
I (1656022890:693) [IoT]: iot_bsp_wifi_set_mode(964) > [rpi] Requested mode SoftAP
I (1656022890:872) [IoT]: _setupHostapd(1345) > [rpi] hostapd config file updated with ssid=Switch Test_E4fnk60016K_ors9an, pw=1111122222, device=wlan0
I (1656022894:272) [IoT]: _SoftAPControl(1282) > [rpi] SoftAP service stop issued
I (1656022896:272) [IoT]: _SoftAPControl(1282) > [rpi] SoftAP service start issued
I (1656022897:414) [IoT]: iot_bsp_wifi_set_mode(1028) > [rpi] AP Mode Started on device wlan0
I (1656022897:414) [IoT]: iot_easysetup_init(494) > IOT_STATE_PROV_ES_START
I (1656022897:414) [IoT]: es_http_init(159) > http tcp init!!
I (1656022897:414) [IoT]: iot_easysetup_init(530) > IOT_STATE_PROV_ES_INIT_DONE
I (1656022897:414) [IoT]: _do_cmd_tout_check(1041) > New state updated for 1
I (1656022897:414) [IoT]: _do_status_report(304) > Call usr status_cb with 2/1
iot_status: 2, lv: 1
E (1656022897:415) [IoT]: http_initialize_connection(234) > Socket unable to bind: errno 98
^C
pi@hostname:~/rpi-st-device $ W (1656022967:709) [IoT]: _do_cmd_tout_check(1028) > New state changing timeout
I (1656022967:709) [IoT]: _do_iot_main_command(498) > curr_main_cmd:10, curr_main_state:1/1
W (1656022967:709) [IoT]: _do_recovery(1429) > state changing fail for 1, curr_state :1
E (1656022967:709) [IoT]: _do_recovery(1488) > Failed process [1] on time, STOP
I (1656022967:710) [IoT]: _do_update_timeout(199) > Current timeout : 4294967295 for 1/-2
I (1656022967:710) [IoT]: _do_status_report(304) > Call usr status_cb with 2/3
iot_status: 2, lv: 3
I (1656022967:710) [IoT]: _do_cmd_tout_check(1041) > New state updated for -2
I (1656022967:710) [IoT]: _do_iot_main_command(498) > curr_main_cmd:10, curr_main_state:-2/-2
W (1656022967:710) [IoT]: _do_state_updating(1711) > Iot-core task will be stopped, needed ext-triggering

I (1656022967:710) [IoT]: http_cleanup_all_connection(39) > close listen socket
I (1656022967:711) [IoT]: es_http_deinit(184) > http tcp deinit complete!
I (1656022967:711) [IoT]: iot_easysetup_deinit(566) > IOT_STATE_PROV_ES_DONE
I (1656022967:711) [IoT]: iot_bsp_wifi_set_mode(851) > [rpi] Requested mode OFF
I (1656022968:824) [IoT]: _SoftAPControl(1282) > [rpi] SoftAP service stop issued
I (1656022969:112) [IoT]: _do_status_report(304) > Call usr status_cb with 1/0
iot_status: 1, lv: 0
I (1656022969:113) [IoT]: _do_update_timeout(199) > Current timeout : 4294967295 for -1/-1
I (1656022969:113) [IoT]: _do_status_report(295) > Unsupported state -1 for 0/0
E (1656022969:122) [IoT]: st_conn_start(1908) > Can't go to PROV_CONFIRM (0x2)
I (1656022969:122) [IoT]: st_conn_start(1910) > st_conn_start done (-7)
```
#### Smartthings Screenshot:
![Screenshot_20220624-002442_SmartThings](https://user-images.githubusercontent.com/32615971/175425254-efebad9a-ba31-48fb-9c68-71c5cba79950.png)

**Comment by toddaustin07** [comment]
This error caused the failure:
```
E (1656022897:415) [IoT]: http_initialize_connection(234) > Socket unable to bind: errno 98
```

That's coming from the core SDK, so not sure what is causing that.

Did you use the mastersetup script to set everything up or did you do any of it manually?  The most frequent problems are with the manual steps relating to the Developer Workspace or, if you did it manually, the configuration of the two json files: onboarding_config.json and device_info.json.   Post the contents of those files here and double check you did everything correctly in the DW - especially copy and pasting the device serial number and key.  If you tried to type those in by hand, that's a potential source for errors.

**Comment by toddaustin07** [comment]
I'm noticing a couple other strange things in the log you posted, which leads me to believe that your wifi seems to have been configured in AP mode when you started the example app.

Please make sure that your wifi is NOT in AP mode.  Maybe it got left in that state when you ran the test.  With the GUI, it's easy to know, but since you don't have that, run this command:
```
iw dev
```
And for your wlan0 device, the type should say 'managed'.  If it says 'AP' you need to reset it.  You can use the resetAP script that is included in my package.  Unless your dhcpcd config files are all messed up, it should be able to reset you back to 'managed' mode for wlan0.  

If the example app fails, you can be left in AP mode - especially if you control-break out of it before the SDK has a chance to shut things down.  So this is another way you might have inadvertently got stuck in AP mode.  Always make sure your wifi is reset before you try to run the example app.  It's possible that this was the cause of the failure, but I'm not positive.  You'll have to try it again.

**Comment by jonas-w** [comment]
It is in managed mode. 
Also restarted the RPI and still no luck.

**Comment by jonas-w** [comment]
Going to test it on my Raspberry PI 4

**Comment by jonas-w** [comment]
Okay this is weird.

This is a problem on my site this time:
My phone can't connect to Smartthings...
![Screenshot_20220624-204259_SmartThings](https://user-images.githubusercontent.com/32615971/175646117-0c8e634d-ec46-4689-8617-98fb8a14fc18.png)

**Comment by jonas-w** [comment]
But with the RPI 3 the other error shows up but not that it can't connect to SmartThings.

Mhm kay will recheck my config on the RPI 4

**Comment by jonas-w** [comment]
Okay sorry for the spam.

Created a new Project and now it works with the Raspberry PI 4

**Comment by toddaustin07** [comment]
Glad you got it working on your Pi4.  

It should also work fine on your Pi3.  Is it a 3B?  What OS version are you running there?

**Comment by jonas-w** [comment]
Yes it's a 3 Model B Rev 1.2.

But probably some misconfiguration in the SmartThings Project.

Because with the PI4 It worked only after creating a fresh Project so maybe the PI3 would work too with a fresh project but I'm ok with using it on my PI4.

But very undescriptive Error Codes on the SmartThings App.

**Comment by toddaustin07** [comment]
Agreed- SmartThings error messages are terrible!
